rpc/src/rpc.rs

```diff
@@ -977,16 +1041,24 @@ impl JsonRpcRequestProcessor {
         }
     }
 
-    fn get_supply(
+    async fn get_supply(
         &self,
         config: Option<RpcSupplyConfig>,
     ) -> RpcCustomResult<RpcResponse<RpcSupply>> {
         let config = config.unwrap_or_default();
         let bank = self.bank(config.commitment);
-        let non_circulating_supply =
-            calculate_non_circulating_supply(&bank).map_err(|e| RpcCustomError::ScanError {
-                message: e.to_string(),
-            })?;
+        let non_circulating_supply = self
+            .runtime
+            .spawn_blocking({
+                let bank = bank.clone();
+                move || {
+                    calculate_non_circulating_supply(&bank).map_err(|e| RpcCustomError::ScanError {
```

Comment:
calculate_non_circulating_supply is called from some other methods, maybe we
should make it async?